### PR TITLE
Parse jason

### PR DESCRIPTION
**Pull request by AngelaLee18**
MVC

Parse Json and activity indicator

DESCRIPTION

In this branch i parse the json file on background and add the activity indicator.

EVIDENCE

Screen Shot 2022-08-24 at 14 29 57

Screen Shot 2022-08-24 at 14 30 52

### CODE REVIEW

File: Ted Talk/Ted Talk/Controller/ViewController.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        }
+        
+        /*func getDataFromBackend(completionHandler: @escaping() -> Void) {
+            DispatchQueue.global(qos: .background).async {
+                self.tableViewData = Parse().parseTedTalk("tedTalks",tableViewData)
+                DispatchQueue.main.async {
+                    completionHandler()
+                }
+            }
+        }*/
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/Controller/ViewController.swift]: there should not be commented code

File: Ted Talk/Ted Talk/Model/Parse.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                    let dataFromJson = try jsonDecoder.decode([TedTalkData].self, from:jsonData )
+                    completion(.success(dataFromJson))
+
+                } catch {
+                    print("Error: \(error)")
+                    completion(.failure(error))
+                }
+            } else {
+                completion(.success([]))
+            }
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/Model/Parse.swift]: do not use if let, instead use guard statement

File: Ted Talk/Ted Talk/Controller/ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    var tableViewData: [TedTalkData] = []
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+        activityIndicator.startAnimating()
+        activityIndicator.hidesWhenStopped = true
+        DispatchQueue.main.asyncAfter(deadline: .now() + 5) {
+            self.activityIndicator.startAnimating()
+        }
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/Controller/ViewController.swift]: the idea of the indicator is show that the data is being parse. There is missing the call to get the TedTalks from Parser.

**Review by ezequielvelezglobant — CHANGES_REQUESTED** [review]
Also there is no test for the parser. Please add it.

File: Ted Talk/Ted Talk/Controller/ViewController.swift (hunk 1 above)
Comment (AngelaLee18) [reply Ted Talk/Ted Talk/Controller/ViewController.swift]: Corrected d7b327bd79c00255aaabf48855a39e1a40b8f5b6

File: Ted Talk/Ted Talk/Model/Parse.swift (hunk 2 above)
Comment (AngelaLee18) [reply Ted Talk/Ted Talk/Model/Parse.swift]: Ok, solved

Comment (AngelaLee18) [reply Ted Talk/Ted Talk/Model/Parse.swift]: 6bc402faf519621cdfbd6c1bde73ffc85e4e9449

File: Ted Talk/Ted Talk/Controller/ViewController.swift (hunk 3 above)
Comment (AngelaLee18) [reply Ted Talk/Ted Talk/Controller/ViewController.swift]: Solved it d7b327bd79c00255aaabf48855a39e1a40b8f5b6

File: Ted Talk/Ted Talk/Controller/ViewController.swift [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+        activityIndicator.hidesWhenStopped = true
+        activityIndicator.startAnimating()
+        DataManager().getDataTedTalks { tedTalksData in
+            self.tableViewData = tedTalksData
+        }
+        DispatchQueue.main.asyncAfter(deadline: .now() + 5) {
+            self.activityIndicator.stopAnimating()
+        }
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/Controller/ViewController.swift]: the stop of the indicator was to be when we get the result of getDataTedTalks

File: Ted Talk/Ted Talk/Model/DataManager.swift [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+//
+//  Created by Angela Lee on 25/08/2022.
+//
+
+import Foundation
+
+public class DataManager {
+    var tedTalks: [TedTalkData] = []
+    func getDataTedTalks(completionHandler: @escaping ([TedTalkData]) -> Void) {
+        Parse().parseTedTalk("tedTalks") { result in DispatchQueue.main.async {
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/Model/DataManager.swift:14]: add a line break before DispatchQueue

**Review by ezequielvelezglobant — CHANGES_REQUESTED** [review]

File: Ted Talk/Ted Talk/Model/DataManager.swift (hunk 5 above)
Comment (AngelaLee18) [reply Ted Talk/Ted Talk/Model/DataManager.swift:14]: 5201db083d82893e742c4f8a3a86f718b3e78d71

File: Ted Talk/Ted Talk/Controller/ViewController.swift (hunk 4 above)
Comment (AngelaLee18) [reply Ted Talk/Ted Talk/Controller/ViewController.swift]: 5201db083d82893e742c4f8a3a86f718b3e78d71

**Review by ezequielvelezglobant — APPROVED** [review]

**Review by JMVarelaGlobant — APPROVED** [review]

**Pull request merged**